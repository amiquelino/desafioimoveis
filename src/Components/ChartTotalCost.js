import React from 'react'
import {Bar} from 'react-chartjs-2'
import Interest from 'interestjs'

const ChartTotalCost = (props) => {

  let rentInterest  = new Interest(props.rentValue, props.dwellingTime, props.annualInterest)
  let buyInterest  = new Interest(props.purchaseValue, props.dwellingTime, props.annualInterest)

  let data = {
    datasets: [
      {
        label: 'Alugar',
        backgroundColor: 'rgba(255,99,132,0.2)',
        borderColor: 'rgba(255,99,132,1)',
        borderWidth: 1,
        hoverBackgroundColor: 'rgba(255,99,132,0.4)',
        hoverBorderColor: 'rgba(255,99,132,1)',
        data: [rentInterest.sum]
      },
      {
        label: 'Comprar',
        backgroundColor: 'rgba(0, 0, 255, 0.3)',
        borderColor: 'rgba(0, 0, 255, 1)',
        borderWidth: 1,
        hoverBackgroundColor: 'rgba(0, 0, 255, 0.4)',
        hoverBorderColor: 'rgba(0, 0, 255, 1)',
        data: [buyInterest.sum]
      }
    ]
  }

  return (
    <div>
      <h2>Custo total</h2>
      <Bar
        data={data}
        width={100}
        height={150}
        options={{
          maintainAspectRatio: false
        }}
      />
    </div>
  )
}

ChartTotalCost.propTypes = {
  rentValue: React.PropTypes.number.isRequired,
  purchaseValue: React.PropTypes.number.isRequired,
  dwellingTime: React.PropTypes.number.isRequired,
  annualInterest: React.PropTypes.number.isRequired
}

export default ChartTotalCost
