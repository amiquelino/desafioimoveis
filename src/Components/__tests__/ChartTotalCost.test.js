import React from 'react'
import renderer from 'react-test-renderer'
import ChartTotalCost from '../ChartTotalCost'

describe('ChartTotalCost', ( ) => {

  it('render using Snapshots', ( ) => {
    let component = renderer.create(
        <ChartTotalCost
          rentValue={3000}
          purchaseValue={100000}
          dwellingTime={10}
          annualInterest={11.5}/>)
    expect( component ).toMatchSnapshot( )
  })
})
