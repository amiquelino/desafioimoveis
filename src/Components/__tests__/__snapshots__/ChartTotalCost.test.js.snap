exports[`ChartTotalCost render using Snapshots 1`] = `
<div>
  <h2>
    Custo total
  </h2>
  <div
    id="chartTotalCost"
    style={
      Object {
        "height": "400px",
        "width": "100%",
      }
    }>
    <div>
      Rendering Chart
    </div>
  </div>
</div>
`;
